src/System.Diagnostics.Contracts/tests/System.Diagnostics.Contracts.Tests.csproj

```diff
@@ -16,6 +16,14 @@
     <None Include="project.json" />
   </ItemGroup>
   <ItemGroup>
+    <!-- Compile tests against the contract, but copy our local-built implementation for testing -->  
+    <ProjectReference Include="..\src\System.Diagnostics.Contracts.csproj">  
```

Comment:
As I called out in the comments I don't think this adds any real value so I wouldn't necessary do this for full facades. Also given that right now all the hacking in the metadata required for full facades. If we do end up needing project references as much of this should be handled by the facade projects themselves and not in the people referencing them.